### PR TITLE
Changes for FindOneAndUpdateCommand and UpdateOneCommand options support

### LINKED ISSUE #161: Implement options for findOneAndUpdate and updateOne
Implement returnDocument and upsert options for findOneAndUpdate command.

Implement upsert options for updateOne command.

Spec is defined in #108

### PR DESCRIPTION
**Pull request by maheshrajamani**
**What this PR does**:
Changes for FindOneAndUpdateCommand and UpdateOneCommand options support

**Which issue(s) this PR fixes**:
Fixes #161 

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [ ] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: src/main/java/io/stargate/sgv2/jsonapi/api/model/command/impl/FindOneAndUpdateCommand.java [hunk 1]
```diff
@@ -17,7 +18,14 @@
...
-    @Nullable Options options)
+    @Valid @Nullable Options options)
     implements ReadCommand, Filterable {
-  public record Options() {}
+  public record Options(
+      @Valid
+          @Nullable
+          @Pattern(
+              regexp = "(after|before)",
+              message = "returnDocument value can only be before or after")
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/api/model/command/impl/FindOneAndUpdateCommand.java:28]: surround before and after in (single?) quotes to indicate they are literal values

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/ReadOperation.java [hunk 2]
```diff
@@ -120,6 +120,9 @@ default Uni<CountResponse> countDocuments(
    */
   Uni<FindResponse> getDocuments(QueryExecutor queryExecutor);
 
+  /** @return */
+  ReadDocument getEmptyDocuments();
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/ReadOperation.java:124]: Javadoc to explain what this is about?

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/FindOperation.java [hunk 3]
```diff
@@ -55,6 +58,22 @@ public Uni<FindResponse> getDocuments(QueryExecutor queryExecutor) {
...
 
+  @Override
+  public ReadDocument getEmptyDocuments() {
+    ObjectNode rootNode = objectMapper().createObjectNode();
+    DocumentId documentId = null;
+    for (DBFilterBase filter : filters) {
+      if (filter instanceof DBFilterBase.IDFilter) {
+        documentId = ((DBFilterBase.IDFilter) filter).value;
+        JsonNode id = ((DBFilterBase.IDFilter) filter).value.asJson(objectMapper());
+        rootNode.putIfAbsent("_id", id);
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/FindOperation.java:69]: This probably works fine, given that `rootNode` is empty Document: we have to ensure `_id` is always the first property.

File: src/main/java/io/stargate/sgv2/jsonapi/service/resolver/model/impl/FindOneAndUpdateCommandResolver.java [hunk 4]
```diff
@@ -35,7 +35,13 @@ public Class<FindOneAndUpdateCommand> getCommandClass() {
   public Operation resolveCommand(CommandContext ctx, FindOneAndUpdateCommand command) {
     ReadOperation readOperation = resolve(ctx, command);
     DocumentUpdater documentUpdater = DocumentUpdater.construct(command.updateClause());
-    return new ReadAndUpdateOperation(ctx, readOperation, documentUpdater, true, shredder);
+    boolean returnUpdatedDocument =
+        command.options() != null
+            && command.options().returnDocument() != null
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/service/resolver/model/impl/FindOneAndUpdateCommandResolver.java:40]: Can avoid null check by switching order of arguments:

   && "after".equals(command.options().returnDocument())

**Review by tatu-at-datastax — APPROVED** [review]

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/FindOperation.java (hunk 3 above)
Comment (maheshrajamani) [reply src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/FindOperation.java:69]: The document from getEmptyDocuments() is used in ReadAndUpdateOperation, where updates part of update clause are applied and Shredder.shred() is called. Shredder takes care of keeping the "_id" as the first field.

File: src/main/java/io/stargate/sgv2/jsonapi/service/resolver/model/impl/FindOneAndUpdateCommandResolver.java (hunk 4 above)
Comment (maheshrajamani) [reply src/main/java/io/stargate/sgv2/jsonapi/service/resolver/model/impl/FindOneAndUpdateCommandResolver.java:40]: Make sense. Changed the conditionality

File: src/main/java/io/stargate/sgv2/jsonapi/api/model/command/impl/FindOneAndUpdateCommand.java (hunk 1 above)
Comment (maheshrajamani) [reply src/main/java/io/stargate/sgv2/jsonapi/api/model/command/impl/FindOneAndUpdateCommand.java:28]: Done.

**Review by tatu-at-datastax — APPROVED** [review]
LGTM

**Review by vkarpov15 — COMMENTED** [review]
LGTM :+1:

**Pull request merged**